chore: readme update (#7)
rudderlabs/rudder-sdk-ruby: README.md

| :--- |

You can then use this client to send the events. A sample `track` call sent using the client is shown below:

```ruby
analytics.track(
  :user_id => '1hKOmRA4GRlm',
  :event => 'Item Sold',
  :properties => { :revenue => 9.95, :shipping => 'Free' }
)
```

## Gzipping requests

The Gzip feature is enabled by default in the Ruby SDK. However, you can disable this feature by setting the `gzip` parameter to false while initializing the SDK:

```ruby
analytics = Rudder::Analytics.new(
  :write_key => 'WRITE_KEY', # required
  :data_plane_url => 'DATA_PLANE_URL',
  :gzip => false, // Set to true to enable Gzip compression
  :on_error => proc { |error_code, error_body, exception, response|
    # defaults to an empty proc
  }
)
```

| Note: Gzip requires `rudder-server` version 1.4 or later. Otherwise, your events might fail. |
| :-----|

## Sending events

Refer to the [RudderStack Ruby SDK documentation](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-ruby-sdk/) for more information on the supported event types.

## Test queue

> You can use the `stub` option to `Rudder::Analytics.new` to cause all requests to be stubbed, making it easier to test with this library.

Reviewer: Yes.

> You can use the `stub` option to `Rudder::Analytics.new` to cause all requests to be stubbed, making it easier to test with this library.

Author: is its default value `true`? https://github.com/rudderlabs/rudder-sdk-ruby/blob/ac142d2f8746e2428f157b162a027eed9bcd8447/spec/rudder/analytics_spec.rb#L8 referencing this. @pallabmaiti

> You can use the `stub` option to `Rudder::Analytics.new` to cause all requests to be stubbed, making it easier to test with this library.

Author: I was wondering if we need this sentence in the readme, we don't have it included in the Ruby docs.

> You can use the `stub` option to `Rudder::Analytics.new` to cause all requests to be stubbed, making it easier to test with this library.

Author: ```suggestion
Enable the `stub` option while initializing the SDK to stub all the requests, making it easier for you to test with this library.
```
